src/Entity/PageManager.php

```diff
@@ -168,4 +139,55 @@ public function getHybridPages(SiteInterface $site): array
             ->getQuery()
             ->execute();
     }
+
+    private function handlePageUrl(PageInterface $page): void
+    {
+        if ($page->isHybrid()) {
+            return;
+        }
+
+        $parent = $page->getParent();
+
+        if (null === $parent) {
+            $page->setSlug(null);
+
+            $url = $page->getCustomUrl() ?? '';
+            $page->setUrl('/'.ltrim($url, '/'));
+
+            return;
+        }
+
+        $slug = $page->getSlug();
+
+        // NEXT_MAJOR: Remove Slugify support
+        if (null === $slug && $this->slugger instanceof Slugify) {
```

Comment:
```suggestion
        if (null === $slug && $this->slugger instanceof SlugifyInterface) {
```
no?

and might be better to write
```suggestion
        if (null === $slug && !$this->slugger instanceof SluggerInterface) {
```